lib/kanal/interfaces/telegram/plugins/telegram_integration_plugin.rb

```diff
@@ -16,38 +16,30 @@ def setup(core)
           end
 
           def register_parameters(core)
-            core.register_input_parameter :tg_message, readonly: true
-            core.register_input_parameter :tg_text, readonly: true
-            core.register_input_parameter :tg_callback, readonly: true
-            core.register_input_parameter :tg_callback_text, readonly: true
             core.register_input_parameter :tg_chat_id, readonly: true
             core.register_input_parameter :tg_username, readonly: true
+            core.register_input_parameter :tg_text, readonly: true
+            core.register_input_parameter :tg_button, readonly: true
```

Comment:
Here you define :tg_button, but at runtime you use :tg_button_pressed

If this is indeed error - rewrite interface so you can test it in spec with fake telegram message object filled with parameters.

Move processing of message from bot.listen do |message| to separate method and remove explicit check for types but check for specific methods instead (or check if !something.nil?) so method can be tested in spec file.

Processing of telegram message and creating kanal input out of it - needs testin in spec file

Reply:
Reverted parameter name change, moved processing to separate method, test for this method created https://github.com/idchlife/kanal-interfaces-telegram/pull/3/commits/cfb7fcc6533ec22a2e882d1b5443a9ea5a8f39bb